### DISCUSSION TITLE
GPU in FINN docker

### DISCUSSION THREAD

**Discussion by btwbtw01** [Xilinx/finn#507] [Q&A]
Hi, 
I want to train VGG16 model using Cifar10 dataset. However, the docker has no GPU to train model, and use CPU to train model is very slow. How can I install GPU in the docker when I have GPU in mt computer?

**Comment by nhma20** [comment]
I also cannot get the GPU to work inside the docker container. Instead, I have a Brevitas/Pytorch environment outside the docker where I can do GPU accelerated training. When done, I simply import the trained network inside the FINN docker environment and continue the workflow there.
Not a solution to your issue, but maybe it's still useful to you.

**Comment by fpjentzsch** [comment]
Hi,
is NVIDIA Container Toolkit installed correctly?

I'm on the dev branch and use FINN from within WSL2. GPU support works fine inside the container, but only if I set `FINN_DOCKER_GPU=1`  manually before running `./run_docker.sh`. The automatic detection (see https://github.com/Xilinx/finn/blob/963aba18db7c42d74b5fa7bd20575a728cbbae01/run-docker.sh#L94) doesn't seem to work in my scenario. Maybe this also works for you?

**Comment by btwbtw01** [comment]
Thanks for your suggestion! Set FINN_DOCKER_GPU=1 doesn't seem to work in my scenario either. I will try @nhma20 solution. Use Brevitas/Pytorch environment outside the docker first, then save model, import the trained network inside the FINN docker.
